rust/Earthfile

```diff
@@ -2,27 +2,36 @@ VERSION 0.7
 
 # CARGO caches the cargo command "cargo $cargo_args"
 # Arguments:
-#   - args (required): Cargo subcommand and its arguments
+#   - args: Cargo subcommand and its arguments. Required
+#   - keep_fingerprints (false): Do no remove source packages fingerprints. Use only when source packages have been COPYed with --keep-ts option
+#   - sweep_days (4): Number of days to retain target build artifacts in cache
 #   - output: Regex to match the files within the target folder to be copied from the cache to the caller filesystem (image layers).
 #     Use this argument when you want to SAVE an ARTIFACT from the target folder (mounted cache), always trying to minimize the total size of the copied fileset.
 #     For example --output="release/[^\./]+" would keep all the files in /target/release that don't have any extension
 CARGO:
     COMMAND
     ARG --required args
     ARG keep_fingerprints=false
+    ARG sweep_days=4
     ARG output
     IF [ "$keep_fingerprints" = "false" ]
        DO +REMOVE_SOURCE_FINGERPRINTS
     END
+    DO +INSTALL_CARGO_SWEEP
     RUN --mount=type=cache,mode=0777,sharing=shared,target=$CARGO_HOME/registry \
         --mount=type=cache,mode=0777,sharing=shared,target=$CARGO_HOME/git \
+        --mount=type=cache,mode=0777,target=/tmp/earthly \
         --mount=type=cache,mode=0777,target=target \
+        echo "Running cargo $args" ; \
         cargo $args; \
         if [ -n "$output" ]; then \
+          echo "Copying output files" ; \
           mkdir /earthly_lib_rust_temp; \
           cd target; \
           find . -type f -regextype posix-egrep -regex "./$output" -exec cp --parents \{\} /earthly_lib_rust_temp \; ; \
-        fi
+        fi; \
+        echo "Running cargo sweep -r -t $sweep_days" ; \
+        /tmp/earthly/cargo-sweep sweep -r -t $sweep_days;
```

Comment:
Could consider `-i` too which will prune artifacts related to toolchains which aren't installed -- likely that would be because the consuming `Earthfile` has been upgraded to use a newer version of rust. OTOH I guess that happens in 4 days anyway and it provides a bit of a footgun for any `Earthfile` which tries to test multiple versions.

Reply:
FYI: 
```
+install-build-dependencies *failed* | Running cargo sweep -dry-run -r -i -t 4
+install-build-dependencies *failed* | error: the argument '--installed' cannot be used with '--time <DAYS>'
```

Reply:
Ah I guess it would need to be run separately then, I think it's perhaps not worth it.

BTW there is also `-s /foo` before and `-f /foo` after, I'm not sure how well that works though, probably too aggressive for the case here.

Reply:
somehow I missed that possibility :)